internal/completions/httpapi/handler.go

```diff
@@ -73,12 +77,27 @@ func newCompletionsHandler(
 			Build()
 		defer done()
 
+		// Use the user's access token for Cody Gateway on dotcom if PLG is enabled
+		accessToken := completionsConfig.AccessToken
+		if envvar.SourcegraphDotComMode() &&
+			featureflag.FromContext(ctx).GetBoolOr("cody-pro", false) &&
+			completionsConfig.Provider == conftypes.CompletionsProviderNameSourcegraph {
+			apiToken, _, err := authz.ParseAuthorizationHeader(r.Header.Get("Authorization"))
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusUnauthorized)
```

Comment:
Careful, if you `http.Error`, you must also return, otherwise execution will continue and you'll make invalid writes on `w`... I've been gotcha'd by this many times 😆 

```suggestion
				http.Error(w, err.Error(), http.StatusUnauthorized)
				return
```

Reply:
I was actually wondering today whether those writes would be blocked and then this would be safe and standard practice. Thanks, Robert, makes total sense!

Reply:
Fixed in 187278c1aec0afbfc77b381cc9f074a4f4443773